drizzlepac/hlautils/diagnostic_json_harvester.py

```diff
@@ -24,11 +25,62 @@
 log = logutil.create_logger(__name__, level=logutil.logging.NOTSET, stream=sys.stdout,
                             format=SPLUNK_MSG_FORMAT, datefmt=MSG_DATEFMT)
 
+# ------------------------------------------------------------------------------------------------------------
+
+
+def filter_header_info(unfiltered_header):
+    """removes unwanted/unneeded keywords from header according to various rules prior to insertion to pandas
+    DataFrame ingest dictionary.
+
+    NOTE: For the time being, this subroutine is an inert placeholder. In the near future, header keywords
+    will be filtered using a keyword whitelist stored in a json harvester settings parameter file.
+
+    Parameters
+    ----------
+    unfiltered_header : dictionary
+        dictionary of unfiltered header information to process
+
+    Returns
+    -------
+    filtered_header : dictionary
+        filtered version of input dictionary *unfiltered_header*
+    """
+    # TODO: IMPLEMENT KEYWORD FILTERING AT A LATER DATE!
+    filtered_header = unfiltered_header
+    return filtered_header
 
 # ------------------------------------------------------------------------------------------------------------
 
 
-def get_json_files(search_path="", log_level=logutil.logging.INFO):
+def flatten_dict(dd, separator='.', prefix=''):
+    """Recursive subroutine to flatten nested dictionaries down into a single-layer dictionary.
+    Borrowed from https://www.geeksforgeeks.org/python-convert-nested-dictionary-into-flattened-dictionary/
+
+    Parameters
+    ----------
+    dd : dict
+        dictionary to flatten
+
+    separator : str, optional
+        separator character used in constructing flattened dictionary key names from multiple recursive
+        elements. Default value is '.'
+
+    prefix : str, optional
+        flattened dictionary key prefix. Default value is an empty string ('').
+
+    Returns
+    -------
+    a version of input dictionary *dd* that has been flattened by one layer
+    """
+    return {prefix + separator + k if prefix else k: v
+            for kk, vv in dd.items()
+            for k, v in flatten_dict(vv, separator, kk).items()
+            } if isinstance(dd, dict) else {prefix: dd}
+
+# ------------------------------------------------------------------------------------------------------------
```

Comment:
Just FYI - there is a Python package, flatten-dict, which will both _flatten_ dict-like objects and _unflatten_ those objects!

Reply:
oh cool! i'll have to look into that.

Reply:
I'll incorporate this into my development towards ticket [HLA-343](https://jira.stsci.edu/browse/HLA-343).